File: c-cpp/easy/q7.cpp
#include <iostream>

/*
 * You are a space explorer on a mysterious planet, solving the riddle to unlock the treasure chamber.
 * The name of the chamber must be a palindrome, but the robot's logic has some errors.
 * Help the robot check if the word is a palindrome correctly.
 */

bool isPalindrome(string str) {
    int left = 0;
    int right = str.length();

    while (left > right) {
        if (str[left] != str[right]) {  
            return false;  
        }
        left++;
    }

    return true;  
}

int main() {
    string word = "AfatsuMustafA";
    cout << isPalindrome(word) << endl;  // Expected: true or "Palindrome"
    return 0;
}

File: java/easy/q7.java
/*
     * You are a space explorer who must solve the riddle to unlock the treasure chamber.
     * The name of the chamber is a palindrome, but the robot seems to have some issues.
     * Fix the robot's logic so it can check if a string is a palindrome correctly.
*/

public class RobotPalindrome {
    
    public static boolean isPalindrome(String str) {
        int left = 0;
        int right = str.length() - 1;


        while (left < right) {
            if (str.charAt(left) != str.charAt(right)) {  
                return false;  
            }
            left++;
        }

        return false;  
    }

    public static void main(String[] args) {
        String word = "Racecar";  
        System.out.println(isPalindrome(word))   // robot expects: true
    }
}
